### PR TITLE
Add the Summarization Playground project

### PR DESCRIPTION
**Pull request by andreban**
This is a demo project for the experimental summarization API that is part of the Early Preview Program for AI APIs in Chrome.

### CODE REVIEW

File: summarization-api-playground/index.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        </div>
+    </main>
+    <footer>
+      Be the first to test new AI APIs. Your feedback is invaluable to our development process. Join our <a href=" https://developer.chrome.com/docs/ai/built-in#get_an_early_preview">Early Preview Program</a> today.
+    </footer>
+    <dialog id="summarization-unavailable">
+      <div>Your browser doesn't support the Summarization API. If you're on Chrome, join the <a href=" https://developer.chrome.com/docs/ai/built-in#get_an_early_preview">Early Preview Program</a> and enable it.</div>
+    </dialog>
+    <dialog id="summarization-unsupported">
+      <div>The Summarization API is available, but your device is unable to run it. Please, check device requirements in the <a href=" https://developer.chrome.com/docs/ai/built-in#get_an_early_preview">Early Preview Program</a> documentation.</div>
```
Comment (tomayac) [inline summarization-api-playground/index.html:36]:
```suggestion
      <div>The Summarization API is available, but your device is unable to run it. Check device requirements in the <a href=" https://developer.chrome.com/docs/ai/built-in#get_an_early_preview">Early Preview Program</a> documentation.</div>
```

File: summarization-api-playground/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * If a downloadProgressCallback is provided, the function will add the callback to the session
+ * creation.
+ * 
+ * The function expects the model availability to be either `readily` or `after-download`, so the
+ * availability must be checked before calling it. If availabilityis `no`, the function will throw
+ *  an error.
+ */
+const createSummarizationSession = async (downloadProgressCallback?: AIModelDownloadCallback): Promise<AISummarizerSession> =>  {
+  const canSummarize = await window.ai.summarizer!.capabilities();
+  if (canSummarize.available == 'no') {
```
Comment (tomayac) [inline summarization-api-playground/src/main.ts:28]:
```suggestion
  if (canSummarize.available === 'no') {
```

File: summarization-api-playground/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    throw new Error('AI Summarization is not supported');
+  }
+
+  const summarizationSession = await window.ai.summarizer!.create();
+  if (canSummarize.available == 'after-download') {
```
Comment (tomayac) [inline summarization-api-playground/src/main.ts:33]:
```suggestion
  if (canSummarize.available === 'after-download') {
```

**Review by tomayac — CHANGES_REQUESTED** [review]
Looks close to good to me.

File: summarization-api-playground/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  let timeout: number | undefined = undefined;
+  inputTextArea.addEventListener('input', () => {
+    characterCountSpan.textContent = inputTextArea.value.length.toFixed();
+    clearTimeout(timeout);
+    timeout = setTimeout(async () => {
+      output.textContent = 'Generating summary...';
+      let session = await createSummarizationSession();
+      let summary = await session.summarize(inputTextArea.value);
+      session.destroy();
```
[the remark this answers is not in the document]
Comment (andreban) [reply summarization-api-playground/src/main.ts:69]: Doing this as per recommendation from eng. So, keeping it for now.

File: summarization-api-playground/src/main.ts (hunk 2 above)
Comment (andreban) [reply summarization-api-playground/src/main.ts:28]: TypeScript does warn for comparison of different types using `==`, but changing for consistency.

File: summarization-api-playground/src/main.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  const canSummarize = await window.ai.summarizer!.capabilities();
+  if (canSummarize.available == 'no') {
+    summarizationUnsupportedDialog.showModal();
+    return;
+  }
+
+  let timeout: number | undefined = undefined;
+  inputTextArea.addEventListener('input', () => {
+    characterCountSpan.textContent = inputTextArea.value.length.toFixed();
+    clearTimeout(timeout);
```
[the remark this answers is not in the document]
Comment (andreban) [reply summarization-api-playground/src/main.ts:64]: Could you explain more? I don't find this particularly confusing, so want to understand better.

File: summarization-api-playground/src/main.ts (hunk 2 above)
Comment (andreban) [reply summarization-api-playground/src/main.ts]: Updated

File: summarization-api-playground/src/main.ts (hunk 3 above)
Comment (andreban) [reply summarization-api-playground/src/main.ts]: Updated

File: summarization-api-playground/src/main.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (tomayac) [reply summarization-api-playground/src/main.ts:64]: It just adds noise. When you wrap it in `debounce()`, it's immediately clear what's happening.

File: summarization-api-playground/src/main.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomayac) [reply summarization-api-playground/src/main.ts:69]: I challenge this. I don't think they mean you should destroy/recreate every second. Probably just that you should destroy at the end of your summarization task.

File: summarization-api-playground/src/main.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (andreban) [reply summarization-api-playground/src/main.ts:64]: I think there's a style preference here - the inline code avoids the reader jumping around to read the debounce function defined elsewhere. I'd agree if the function within `setTimeout` was larger, but a debounce here will reduce clarity by introducing indirection.

File: summarization-api-playground/src/main.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (andreban) [reply summarization-api-playground/src/main.ts:69]: This is **not** destroying/creating every second (this is `setTimeout()`, not `setInterval()`). What this does is create a timeout when the user types to summarize the content after 1 second. If the user types more within that second, it clears the timeout, preventing the scheduled summarization from running, and creates a new one for 1 second later. Effectively, this means the content of the textarea is summarized when the user hasn't typed for at least 1 second.

[the remark this answers is not in the document]
Comment (tomayac) [reply summarization-api-playground/src/main.ts:69]: Duh, I was reviewing on mobile for my excuse. You're right. But also: _qed_. I got confused. No one would read the `debounce()` implementation. It's just a utility with an immediately clear name.

[the remark this answers is not in the document]
Comment (tomayac) [reply summarization-api-playground/src/main.ts:69]: Pull in something like https://lodash.com/docs/#debounce and be good IMHO.

[the remark this answers is not in the document]
Comment (andreban) [reply summarization-api-playground/src/main.ts:72]: Clarified in a code comment.

File: summarization-api-playground/src/main.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (tomayac) [reply summarization-api-playground/src/main.ts:67]: This works for me. Resolving.

File: summarization-api-playground/src/main.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomayac) [reply summarization-api-playground/src/main.ts:72]: Resolving as per the other comment.

**Review by tomayac — APPROVED** [review]
LGTM, thanks for working on this!

**Pull request merged**